lib/charms/kubernetes_charm_libraries/v0/kubernetes_hugepages_volumes_patch.py

```diff
@@ -0,0 +1,624 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charm library used to manage HugePages volumes in Kubernetes charms.
+
+- On bound event (e.g. self.on.hugepages_volumes_config_changed which is originated
+from K8sHugePagesVolumePatchChangedEvent), it will:
```

Comment:
this is not super clear to me, what we want to convey is that charm authors can use whatever event they choose to trigger refresh